internal/manifest/list.go

```diff
@@ -65,10 +65,12 @@ type List interface {
 // ListUpdate includes the fields which a List's UpdateInstances() method will modify.
 // This is publicly visible as c/image/manifest.ListUpdate.
 type ListUpdate struct {
-	Digest    digest.Digest
-	Size      int64
-	MediaType string
-	Platform  *imgspecv1.Platform // read-only field: may be set by Instance(), ignored by UpdateInstance()
+	Digest                    digest.Digest
+	Size                      int64
+	MediaType                 string
+	Platform                  *imgspecv1.Platform // read-only field: may be set by Instance(), ignored by UpdateInstance()
```

Comment:
Consider moving the read-only fields into an inner struct:
```golang
type ListUpdate struct {
    [...]
    struct ReadOnly {
        Platform string
        [...]
    }
}
```

This way, all users have to access it via `(ListUpdate).ReadOnly.Platform` which is quite expressive.

Reply:
Nice Idea, Done